src/object_functions.c

```diff
@@ -0,0 +1,2016 @@
+GLOBAL_ASM("asm/non_matchings/unknown_032760/obj_loop_effectbox.s")
+#endif
+
+void obj_init_trophycab(Object *obj, LevelObjectEntry_TrophyCab *entry) {
+    obj->unk4C->unk14 = 1;
+    obj->unk4C->unk11 = 2;
+    obj->y_rotation = entry->rotation << 6 << 4; // Not sure about the values here.
+}
+
+#ifdef NON_EQUIVALENT
+
+void func_80001D04(u16, s32 *);
+f32 sqrtf(f32);
+void func_80001D04(u16, s32 *);
+f32 sqrtf(f32);
+u32 func_80001C08(void);
+void func_80001D04(u16, s32 *);
+Object *func_8000EA54(unk80027FC4 *, s32);
+void func_8005A3B0(void);
+void func_8006F254(void);
+void func_8009CF68(s32 arg0);
+s32 func_8009CFEC(u32 arg0);
+void func_800A3870(void);
+void func_800AB1AC(s32 arg0);
+void func_800AB1D4(u8 arg0);
+void func_800C31EC(s32);
+s32 func_800C3400(void);
+LevelHeader *get_current_level_header(void);
+Settings *get_settings(void);
+void play_sound(u8 arg0);
+void set_music_fade_timer(s32 time);
+void set_sndfx_player_voice_limit(u8 voiceLimit);
+f32 sqrtf(f32);
+
+typedef struct Object_64_80034E9C {
+    s32 unk0;
+    s16 unk4;
+} Object_64_80034E9C;
+
+typedef struct Object_54_80034E9C {
+    u8 pad0[0x28];
+    f32 unk28;
+    f32 unk2C;
+} Object_54_80034E9C;
+
+// Has regalloc & stack issues.
+
+void obj_loop_trophycab(Object *obj, s32 speed) {
+    s32 sp34;
+    s32 isTrophyRaceAvaliable;
+    s32 raceType;
+    unk80027FC4 sp44;
+    Settings *settings;
+    LevelHeader *curLevelHeader;
+    Object_64_80034E9C *obj64;
+    Object *trophyObj;
+    Object *playerObj;
+    f32 xDiff;
+    f32 zDiff;
+    s32 new_var2;
+    
+    settings = get_settings();
+    curLevelHeader = get_current_level_header();
+    obj64 = obj->unk64;
+    
+    // Show trophy inside the cabinet
+    if (obj->unk7C.word == 0) {
+        raceType = curLevelHeader->race_type;
+        if ((raceType != 7) && (raceType != 6)) { // Make sure the current level is not a cutscene.
+            obj->unk7C.word = 1;
+            if ((settings->trophies >> (((settings->worldId - 1) ^ 0) * 2)) & 3) {
+                sp44.unk0 = 128;
+                sp44.unk2 = obj->unk3C_a.unk3C->unk2;
+                sp44.unk4 = obj->unk3C_a.unk3C->unk4;
+                sp44.unk6 = obj->unk3C_a.unk3C->unk6;
+                sp44.unk1 = 8;
+                trophyObj = func_8000EA54(&sp44, 1);
+                if (trophyObj) {
+                    trophyObj->unk3C_a.unk3C = (void *) 0;
+                    trophyObj->y_rotation = obj->y_rotation;
+                }
+            }
+        }
+    }
+    
+    ((Object_54_80034E9C *) obj->unk54)->unk2C = 0.612f;
+    ((Object_54_80034E9C *) obj->unk54)->unk28 = 0.0f;
+    
+    playerObj = get_object_struct(0);
+    if (playerObj) {
+        xDiff = obj->x_position - playerObj->x_position;
+        zDiff = playerObj->z_position - obj->z_position;
+        sqrtf((xDiff * xDiff) + (zDiff * zDiff)); // Distance on X & Z axes. Not used?
+    
+        new_var2 = settings->bosses;
+        new_var2 |= 0x800; // Why OR with 0x800?
```

Comment:
0x800 is the FFL boss rematch, which doesn't exist. This is necessary so `isTrophyRaceAvaliable` is true once all the balloons in FFL (worldId 5) are collected, (i.e. `((1 << (settings->worldId + 6)) & new_var2) != 0`).